### ISSUE TITLE
Apollo Server Lambda 2 sometimes has unhandled promise exception on error

### ISSUE THREAD

**Issue by stubailo** [apollographql/apollo-server#1289]
I don't have a simple repro or failing test other than this whole file yet: https://github.com/stubailo/pupstagram-netlify-functions/blob/a0763a17e764dd3780a2dd4b22402649ccd8db60/src/lambda/graphql.js

It's on this line, when trying to read `result.headers`, AS has an unhandled exception since `result` is undefined: https://github.com/apollographql/apollo-server/blob/19478920e4a6bdbf57b10ddc782e685542fd9bd4/packages/apollo-server-lambda/src/ApolloServer.ts#L127

Here's the output from Netlify functions (aka lambda):

```
4:26:13 PM: graphql invoked
4:26:14 PM: (node:1) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'headers' of undefined
    at callbackFilter (/var/task/graphql.js:110438:95)
    at apollo_server_core_1.runHttpQuery.then (/var/task/graphql.js:110551:24)
    at <anonymous>
4:26:14 PM: (node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
4:26:14 PM: (node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
4:26:23 PM: Task timed out after 10.00 seconds
```

**Comment by stubailo** [comment]
cc @adnsio any ideas about a simple fix here?

**Comment by stubailo** [comment]
Evans said:

> Looks like we just need to add an existence check on result before passing the spread object to the callback, since there’s a chance that the result doesn’t exist https://github.com/apollographql/apollo-server/blob/19478920e4a6bdbf57b10ddc782e685542fd9bd4/packages/apollo-server-lambda/src/lambdaApollo.ts#L59

**Comment by adnsio** [comment]
@evans has released a fix in this commit: https://github.com/apollographql/apollo-server/commit/035175144a8751660f6f52f178f5ecceab3d4c44

**Comment by evans** [comment]
Solved with apollo-server-lambda@2.0.0-rc.7
